ledger-core/v2/application/functest/wallet_scenarios_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+// +build functest
+
+package functest
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateUpdateWallet(t *testing.T) {
+	var ref string
+
+	t.Log("1.Create wallet")
+	{
+		walletRef, err := createSimpleWallet()
+		require.NoError(t, err)
+
+		ref = walletRef
+		require.NotEmpty(t, ref, "wallet reference mustn't be empty")
+	}
+	t.Log("2.Get wallet balance")
+	{
+		getBalanceURL := getURL(walletGetBalancePath, "", "")
+		balance, err := getWalletBalance(getBalanceURL, ref)
+		require.NoError(t, err)
+		require.Equal(t, 1000, balance, "wrong balance amount")
+	}
+	t.Log("3.Add amount to wallet")
+	{
+		addAmountURL := getURL(walletAddAmountPath, "", "")
+		err := addAmountToWallet(addAmountURL, ref, 100)
+		require.NoError(t, err)
+	}
+	t.Log("4.Get wallet balance")
+	{
+		getBalanceURL := getURL(walletGetBalancePath, "", "")
+		balance, err := getWalletBalance(getBalanceURL, ref)
+		require.NoError(t, err)
+		require.Equal(t, 1100, balance, "wrong balance amount")
+	}
+}
+
+func TestGetUpdateBalanceConcurrently(t *testing.T) {
+	var (
+		ref              string
+		count            = 10 // Number of concurrent requests per node.
+		amount           = 100
+		expectedBalance  = 1000 + amount*count*len(nodesPorts)
+		goroutineCounter = 0
+		resultErr        error
+		outChan          = make(chan error)
+	)
+
+	t.Log("1.Create wallet")
+	{
+		walletRef, err := createSimpleWallet()
+		require.NoError(t, err)
+
+		ref = walletRef
+		require.NotEmpty(t, ref, "wallet reference mustn't be empty")
+	}
+	t.Log("2.Concurrent requests to /add_amount and /get_balance")
+	{
+		resultCollector := func() {
```

Comment:
why do you need recover panic?

Reply:
It's panic in utility goroutine, why won't it be recovered and translated to error?

Reply:
I mean, which method raises panic?

Reply:
For example, `getWalletBalance`->`sendAPIRequest`->`doReq`->`ioutil.ReadAll` can raise panic

Reply:
hmmm, okay